database/database.go

```diff
@@ -170,6 +185,103 @@ func (db SqlDatabase) AddImage(uuid string, name string, alt string) (err error)
 	return nil
 }
 
+func (db SqlDatabase) AddCard(uuid string, image_location string, json_data string, schema_name string) (err error) {
+	// Check that the file exists and is a file
+	// not a directory. Ideally, check the ext
+	if image_location == "" {
+		return fmt.Errorf("cannot have image")
+	}
+	image_stat, err := os.Stat(image_location)
+	if errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("file does not exist: %s", image_location)
+	}
+	if err != nil {
+		return err
+	}
+	if image_stat.IsDir() {
+		return fmt.Errorf("given path is a directory: %s", image_location)
+	}
+
+	// Load the schema
+	// TODO : probably pass the schema data instead
+	if json_data == "" {
+		return fmt.Errorf("cannot have empty data")
+	}
+
+	if schema_name == "" {
+		return fmt.Errorf("cannot have an empty schema name")
+	}
+
+	_, err = validateJson(json_data, schema_name)
+	if err != nil {
+		return err
+	}
```

Comment:
All this validation should happen in the service. The database should only persist the valid data to the database. Separating business and infrastructure (i.e. database operations) logic is key to having a clean architecture.

Reply:
I need to continue working on this. I did some poc work during a stream and forgot to finish it off!